RecoHGCal/TICL/plugins/TrackstersToMultiClusterProducer.cc

```diff
@@ -0,0 +1,92 @@
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/ParticleFlowReco/interface/HGCalMultiCluster.h"
+#include "RecoHGCal/TICL/interface/Trackster.h"
+
+#include <algorithm>
+#include <array>
+#include <string>
+#include <vector>
+
+class TrackstersToMultiClusterProducer : public edm::stream::EDProducer<> {
+ public:
+  TrackstersToMultiClusterProducer(const edm::ParameterSet&);
+  ~TrackstersToMultiClusterProducer() override {}
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+ private:
+  std::string label_;
+  edm::EDGetTokenT<std::vector<reco::CaloCluster>> layer_clusters_token_;
+  edm::EDGetTokenT<std::vector<Trackster>> tracksters_token_;
+};
+
+DEFINE_FWK_MODULE(TrackstersToMultiClusterProducer);
+
+TrackstersToMultiClusterProducer::TrackstersToMultiClusterProducer(const edm::ParameterSet& ps)
+    : label_(ps.getParameter<std::string>("label")),
+  layer_clusters_token_(consumes<std::vector<reco::CaloCluster>>(ps.getParameter<edm::InputTag>("LayerClusters"))),
+  tracksters_token_(consumes<std::vector<Trackster>>(ps.getParameter<edm::InputTag>("Tracksters"))) {
+  produces<std::vector<reco::HGCalMultiCluster>>(label_);
+}
+
+void TrackstersToMultiClusterProducer::fillDescriptions(
+    edm::ConfigurationDescriptions& descriptions) {
+  // hgcalMultiClusters
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("Tracksters", edm::InputTag("Tracksters", "TrackstersByCA"));
+  desc.add<edm::InputTag>("LayerClusters", edm::InputTag("hgcalLayerClusters"));
+  desc.add<std::string>("label", "MultiClustersFromTracksterByCA");
+  desc.addUntracked<unsigned int>("verbosity", 3);
+  descriptions.add("TrackstersToMultiCluster", desc);
```

Comment:
please start module instance name from a lower case.
Consider to flip the order in the name: ```MultiClustersFromTrackstersProducer``` to be more clear what is made rather than the origin.